react/index.d.ts

```diff
@@ -37,7 +37,7 @@ declare namespace React {
 
     type CElement<P, T extends Component<P, ComponentState>> = ComponentElement<P, T>;
     interface ComponentElement<P, T extends Component<P, ComponentState>> extends ReactElement<P> {
-        type: ComponentClass<P>;
+        type: ComponentClass<P> | ComponentClass<any>;
```

Comment:
Can you give an example? Other top-level API methods rely on this being strongly typed (eg. to infer the correct return type of `ReactDOM.render`) so I'm worried that the `any` will break that.